merkletree.go

```diff
@@ -80,12 +93,13 @@ func lookUp(key string, str *SignedTreeRoot) (MerkleNode, []MerkleNode, error) {
 	}
 	switch nodePointer.(type) {
 	case *userLeafNode:
+		proof = append(proof, nodePointer.(*userLeafNode))
 		if nodePointer.(*userLeafNode).key == key {
-			proof = append(proof, nodePointer.(*userLeafNode))
 			return nodePointer.(*userLeafNode), proof, nil
-		} else {
-			// TODO: what should we return here?
 		}
+		// reached a different leaf with a matching prefix
+		// return nil and a auth path including the leaf node
+		return nil, proof, nil
```

Comment:
Since the proof includes the different leaf, the verifier could still see all the contents of the leaf, including the key and the value, right? Or am I missing something? Shouldn't there be a step here that "removes" the ```key``` and the ```value``` from the leaf in the proof to preserve privacy?